tests/databases/test_snowflake.py

```diff
@@ -153,6 +153,38 @@ def test_load_pandas_dataframe_to_table(database_table_fixture):
     assert rows[1] == (2,)
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.SNOWFLAKE,
+            "table": Table(metadata=Metadata(schema=SCHEMA)),
+        },
+    ],
+    indirect=True,
+    ids=["snowflake"],
+)
+def test_if_exist_param_of__load_pandas_dataframe_to_table(database_table_fixture):
+    """Test if_exist parameter on load_pandas_dataframe_to_table against snowflake"""
+    database, table = database_table_fixture
+
+    pandas_dataframe = pd.DataFrame(data={"id": [1, 2]})
+
+    with mock.patch("snowflake.connector.pandas_tools.write_pandas"):
+        with mock.patch("astro.databases.base.BaseDatabase.create_table") as method:
+            database.load_pandas_dataframe_to_table(
+                pandas_dataframe, table, if_exists="replace"
+            )
+            assert method.called
+
+        with mock.patch("astro.databases.base.BaseDatabase.create_table") as method:
+            database.load_pandas_dataframe_to_table(
+                pandas_dataframe, table, if_exists="append"
+            )
+            assert not method.called
```

Comment:
Did you verify that this test fails without this change?

Reply:
Yes, I was in a call with @bharanidharan14, when this was failing on multiple occasions.